### PR TITLE
Github actions for nightly check db

### PR DESCRIPTION
**Pull request by lyhe18**
Add nightly check db yaml to this branch. Including the actions of download dataset, move data and check db.

### CODE REVIEW

File: .github/workflows/nightly-check-db.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    - name: Download db snapshot
+      env:
+        PYTHONPATH: /tmp
+      run: |
+        mkdir -p workspace
+        curl -s https://s3-us-west-2.amazonaws.com/qkcmainnet/data/sample-data.tar.gz --output data.tar.gz
+        tar xvfz data.tar.gz
+        rm data.tar.gz && mv data workspace/
+
+    - name: Save cache
```
Comment (ninjaahhh) [inline .github/workflows/nightly-check-db.yml]: can skip saving/loading cache and use a single job

File: .github/workflows/nightly-check-db.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    runs-on: ubuntu-latest
+    container: quarkchaindocker/pyquarkchain:mainnet1.1.1
+    
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Download db snapshot
+      env:
+        PYTHONPATH: /tmp
```
Comment (ninjaahhh) [inline .github/workflows/nightly-check-db.yml]: hmm I don't think PYTHON is needed here. do you mean current working directory?

File: .github/workflows/nightly-check-db.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      run: |
+        mkdir -p workspace
+        curl -s https://s3-us-west-2.amazonaws.com/qkcmainnet/data/sample-data.tar.gz --output data.tar.gz
+        tar xvfz data.tar.gz
+        rm data.tar.gz && mv data workspace/
+
+    - name: Move data
+      run: |
+        mkdir -p quarkchain/cluster/qkc-data
+        cd /tmp/workspace && mv data mainnet && cd -
```
Comment (ninjaahhh) [inline .github/workflows/nightly-check-db.yml]: I don't think data is persisted into `/tmp/workspace`

**Comment by ninjaahhh** [comment]
why do you still have `.circle`? did you merge master branch?

File: .github/workflows/nightly-check-db.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  download-snapshaot-move-data-check-db:
+
+    runs-on: ubuntu-latest
+    container: quarkchaindocker/pyquarkchain:mainnet1.1.1
+    
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Download db snapshot
+      working-directory: /temp
```
Comment (ninjaahhh) [inline .github/workflows/nightly-check-db.yml]: can combine downloading and moving data to a single step

Comment (lyhe18) [reply .github/workflows/nightly-check-db.yml]: ok~

File: .github/workflows/nightly-check-db.yml (hunk 2 above)
Comment (lyhe18) [reply .github/workflows/nightly-check-db.yml]: yes, I have used the keyword of working directory instead.

File: .github/workflows/nightly-check-db.yml (hunk 4 above)
Comment (lyhe18) [reply .github/workflows/nightly-check-db.yml]: Done.

File: .github/workflows/nightly-check-db.yml (hunk 1 above)
Comment (lyhe18) [reply .github/workflows/nightly-check-db.yml]: Done.

File: .github/workflows/nightly-check-db.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        mv /tmp/workspace/mainnet quarkchain/cluster/qkc-data/
+
+    - name: Build qkchash lib
+      run: |
+        cd qkchash && make clean && make && cd -
+
+    - name: check db
+      run: |
+        height=$(python quarkchain/tools/db_browser.py --cluster_config=`pwd`/mainnet/singularity/cluster_config_template.json root_print_tip 2> /dev/null | grep "height" | awk "{print \$2}" | sed "s/,$//")
+        lo=$(( height / CIRCLE_NODE_TOTAL * CIRCLE_NODE_INDEX ))
```
Comment (ninjaahhh) [inline .github/workflows/nightly-check-db.yml]: note we are not in circleci anymore

Comment (lyhe18) [reply .github/workflows/nightly-check-db.yml]: Done.

File: .github/workflows/nightly-check-db.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Download db snapshot and move data
+      run: |
+        mkdir -p /tmp/workspace &&  cd /tmp
+        curl -s https://s3-us-west-2.amazonaws.com/qkcmainnet/data/sample-data.tar.gz --output data.tar.gz
+        tar xvfz data.tar.gz
+        rm data.tar.gz && mv data workspace/ && cd -
```
Comment (ninjaahhh) [inline .github/workflows/nightly-check-db.yml]: you can skip those intermediary directories

**Comment by ninjaahhh** [comment]
also, use a more explicit title for the PR

**Pull request closed without merging**